Sources/TensorFlow/Layers/Normalization.swift

```diff
@@ -70,48 +70,33 @@ public struct BatchNorm<Scalar: TensorFlowFloatingPoint>: Layer {
         let positiveAxis = (input.rank + axis) % input.rank
         var normalizedAxes = Array(0..<input.rank)
         normalizedAxes.remove(at: positiveAxis)
-        let mean = input.mean(alongAxes: normalizedAxes)
-        let variance = input.variance(alongAxes: normalizedAxes)
-        runningMean.value += (mean - runningMean.value) * (1 - momentum)
-        runningVariance.value += (variance - runningVariance.value) * (1 - momentum)
-        let inv = rsqrt(variance + epsilon) * scale
-        return (input - mean) * inv + offset
+        let moments = input.moments(alongAxes: normalizedAxes)
+        runningMean.value += (moments.mean - runningMean.value) * (1 - momentum)
+        runningVariance.value += (moments.variance - runningVariance.value) * (1 - momentum)
+        let inv = rsqrt(moments.variance + epsilon) * scale.reshaped(to: moments.variance.shape)
+        return (input - moments.mean) * inv + offset.reshaped(to: moments.mean.shape)
     }
 
     @differentiable
     private func applyingInference(to input: Tensor<Scalar>) -> Tensor<Scalar> {
-        let inv = rsqrt(runningVariance.value + epsilon) * scale
-        return (input - runningMean.value) * inv + offset
+        let scaleShape = runningVariance.value.shape
+        let offsetShape = runningMean.value.shape
+        let inv = rsqrt(runningVariance.value + epsilon) * scale.reshaped(to: scaleShape)
+        return (input - runningMean.value) * inv + offset.reshaped(to: offsetShape)
     }
 
     /// Returns the output obtained from applying the layer to the given input.
     ///
     /// - Parameter input: The input to the layer.
     /// - Returns: The output.
-    @differentiable(vjp: _vjpApplied(to:))
```

Comment:
Control flow differentiation is known to have memory leaks (#364).  Though it's going to be fixed some time this week, I'd suggest making this change in a later PR instead.

Reply:
I added back _vjpApplied() as suggested.

Reply:
Could you please revert the change so that the original indentation won't get modified?

Reply:
I have reverted to the original for these lines now.